crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -219,12 +219,31 @@ def services
         # Initiate the services shutdown for all nodes
         errors = []
         upgrade_nodes = NodeObject.find("state:crowbar_upgrade")
-        upgrade_nodes.each do |upgrade_node|
-          cmd = upgrade_node.shutdown_services_before_upgrade
+        cinder_nodes = []
+        upgrade_nodes.each do |node|
+          if node.roles.include?("cinder-controller") &&
+              (!node.roles.include?("pacemaker-cluster-member") || node["pacemaker"]["founder"])
+            cinder_nodes.push node
+          end
+          cmd = node.shutdown_services_before_upgrade
           next if cmd[0] == 200
           errors.push(cmd[1])
         end
 
+        cinder_nodes.each do |node|
```

Comment:
I would expect the list will have just one item, yes. Can't there be more clusters and thus more controllers in different clusters? (Probably not)